SDWebImage/NSImage+WebCache.m

```diff
@@ -28,10 +28,8 @@ - (BOOL)isGIF {
         if ([rep isKindOfClass:[NSBitmapImageRep class]]) {
             NSBitmapImageRep *bitmapRep = (NSBitmapImageRep *)rep;
             NSUInteger frameCount = [[bitmapRep valueForProperty:NSImageFrameCount] unsignedIntegerValue];
-            if (frameCount > 1) {
-                isGIF = YES;
-                break;
-            }
+            isGIF = frameCount > 1 ? YES : NO;
```

Comment:
NSImage 's design is that it use NSImageRep for actually rendering images. It choose the "best represent" to showing. In fact, GIF is bitmapRep without scale hint for PDF or someother things. So we do not need to check all image representations.